src/api/app/controllers/webui/package_controller.rb

```diff
@@ -1037,6 +1023,14 @@ def binary_download
 
   private
 
+  def validate_xml
```

Comment:
This would be called before the authorization now. Probably not a big deal, but would be nicer to have the authorization before the validation. I guess moving this to the model an have the validation there, like a normal active record validation, would require to much refactoring:thinking:

Reply:
it would, you know that :japanese_ogre:

Reply:
BTW, the XML parsing is already being called before the authentication https://github.com/openSUSE/open-build-service/blob/master/src/api/app/controllers/webui/package_controller.rb#L983